src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
Another way to look at it: Making registry trump the loader means that the targetname namespace is essentially a global namespace.  Making loader trump registry means that empty modulename namespace is just *one* of the possible module namespaces.

If we have two modules that provide the same targetnames, how do we override both at the same time?  We can't if the targetname trumps loader. In fact, overriding one automatically overrides the other.  That seems problematic, if we think of the registry as an override for module namespaces.